src/scheduler/limits.cpp

```diff
@@ -1501,20 +1509,20 @@ check_queue_max_user_run(server_info *si, queue_info *qi, resource_resv *rr,
 	limcounts *sc, limcounts *qc, schd_error *err)
 {
 	char		*key;
-	char		*user = rr->user;
+	std::string	user = rr->user;
 	int		used;
 	int		max_user_run, max_genuser_run;
 	counts		*cts = NULL;
 
-	if ((qi == NULL) || (user == NULL) || (qc == NULL))
+	if ((qi == NULL) || (user == empty_str) || (qc == NULL))
 		return (SCHD_ERROR);
 
 	if (!qi->has_user_limit)
 	    return (0);
 
 	cts = qc->user;
 
-	if ((key = entlim_mk_runkey(LIM_USER, user)) == NULL)
+	if ((key = entlim_mk_runkey(LIM_USER, user.c_str())) == NULL)
```

Comment:
That's not possible, it is in libutils that's common across daemons.

Reply:
Depending how often you have to call out user.c_str(), you can make a local overload.  This is what I did for log_eventf(), so it could take a string.